src/main/java/org/jboss/modules/JDKSpecific.java

```diff
@@ -64,7 +63,9 @@
     // === the actual JDK-specific API ===
 
     static Class<?> getCallingUserClass() {
-        return STACK_WALKER.walk(stream -> stream.skip(1).findFirst().get().getDeclaringClass());
+        return STACK_WALKER.walk(stream -> stream.skip(1)
+                .filter(s -> !s.getDeclaringClass().equals(org.jboss.modules.Module.class))
+                .findFirst().get().getDeclaringClass());
```

Comment:
That problem pops up in WildFly-Core test suite. Here's sample stack trace:

^[[0m^[[31m16:08:46,584 ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0013: Operation ("parallel-extension-add") failed - address: ([]): java.lang.RuntimeException: WFLYCTL
<------>at org.jboss.as.controller//org.jboss.as.controller.extension.ParallelExtensionAddHandler$1.execute(ParallelExtensionAddHandler.java:115)
<------>at org.jboss.as.controller//org.jboss.as.controller.AbstractOperationContext.executeStep(AbstractOperationContext.java:940)
<------>at org.jboss.as.controller//org.jboss.as.controller.AbstractOperationContext.processStages(AbstractOperationContext.java:683)
<------>at org.jboss.as.controller//org.jboss.as.controller.AbstractOperationContext.executeOperation(AbstractOperationContext.java:382)
<------>at org.jboss.as.controller//org.jboss.as.controller.OperationContextImpl.executeOperation(OperationContextImpl.java:1363)
<------>at org.jboss.as.controller//org.jboss.as.controller.ModelControllerImpl.boot(ModelControllerImpl.java:477)
<------>at org.jboss.as.controller//org.jboss.as.controller.AbstractControllerService.boot(AbstractControllerService.java:390)
<------>at org.jboss.as.controller//org.jboss.as.controller.AbstractControllerService.boot(AbstractControllerService.java:352)
<------>at org.jboss.as.server//org.jboss.as.server.ServerService.boot(ServerService.java:395)
<------>at org.jboss.as.server//org.jboss.as.server.ServerService.boot(ServerService.java:364)
<------>at org.jboss.as.controller//org.jboss.as.controller.AbstractControllerService$1.run(AbstractControllerService.java:302)
<------>at java.base/java.lang.Thread.run(Thread.java:843)
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException
<------>at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
<------>at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
<------>at org.jboss.as.controller//org.jboss.as.controller.extension.ParallelExtensionAddHandler$1.execute(ParallelExtensionAddHandler.java:107)
<------>... 11 more
Caused by: java.lang.NullPointerException
<------>at org.jboss.modules.Module.loadServiceFromCallerModuleLoader(Module.java:443)
<------>at org.jboss.modules.Module.loadServiceFromCallerModuleLoader(Module.java:427)
<------>at org.jboss.as.controller//org.jboss.as.controller.extension.ExtensionAddHandler.initializeExtension(ExtensionAddHandler.java:120)
<------>at org.jboss.as.controller//org.jboss.as.controller.extension.ExtensionAddHandler.initializeExtension(ExtensionAddHandler.java:104)
<------>at org.jboss.as.controller//org.jboss.as.controller.extension.ParallelExtensionAddHandler$ExtensionInitializeTask.call(ParallelExtensionAddHandler.java:144)
<------>at org.jboss.as.controller//org.jboss.as.controller.extension.ParallelExtensionAddHandler$ExtensionInitializeTask.call(ParallelExtensionAddHandler.java:127)
<------>at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
<------>at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1161)
<------>at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
<------>at java.base/java.lang.Thread.run(Thread.java:843)
<------>at org.jboss.threads//org.jboss.threads.JBossThread.run(JBossThread.java:320)